packages/ozone-helper/ozone-api-request/src/ozone-api-request.ts

```diff
@@ -75,182 +74,190 @@ export type OzoneAPIRequestOption = {
-     * Default value is document.
-     * @type {EventTarget}
-     */
-    eventTarget: EventTarget | null;
-    setEventTarget(element: EventTarget){
-        this.eventTarget = element;
-    }
-
-    private dispatchEvent(eventName: string, xmlhttp: XMLHttpRequest){
-        if(this.eventTarget){
-            this.eventTarget.dispatchEvent(new CustomEvent<XMLHttpRequest>(eventName, {
-                bubbles: true, detail: xmlhttp
-            }));
-        }
-    }
-
-    /**
-     *
-     * @param {XMLHttpRequest} request (optional) This parameters overwrite the default XmlHttpRequest.
-     * @return {OzoneAPIRequest}
-     */
-    send(request?: XMLHttpRequest):OzoneAPIRequest{
-        const xmlhttp = this.currentRequest = request || this.createXMLHttpRequest();
-
-        if(this._onreadystatechange)
-            xmlhttp.onreadystatechange = this._onreadystatechange;
-
-
-        this._resultPromise = new Promise((resolve, reject)=>{
-            this.resolveCurrentRequest = resolve;
-
-            const handleResponse = ()=>{
-                if (xmlhttp.status >= 200 && xmlhttp.status < 300) {
-                    this.dispatchEvent('ozone-api-request-success', xmlhttp);
-                    resolve(xmlhttp);
-                } else if(xmlhttp.status === 403) {
-                    this.dispatchEvent('ozone-api-request-unauthorized', xmlhttp);
-                    reject(xmlhttp);
-                } else {
-                    this.dispatchEvent('ozone-api-request-error', xmlhttp);
-                    reject(xmlhttp);
-                }
-            };
-
-            xmlhttp.onload = handleResponse;
-
-            xmlhttp.ontimeout = () =>{
-                this.dispatchEvent('ozone-api-request-timeout',  xmlhttp);
-                reject(xmlhttp);
-            };
-
-            xmlhttp.onerror = handleResponse;
-
-            xmlhttp.send(this.body);
-        });
-        return this;
-    }
-
-    /**
-     *
-     * @param {XMLHttpRequest} request (optional) This parameters overwrite the default XmlHttpRequest.
-     * @return {Promise<XMLHttpRequest>}
-     */
-    sendRequest(request?: XMLHttpRequest):Promise<XMLHttpRequest>{
-        return this.send(request).result
-
-    }
+export class OzoneAPIRequest {
+
+	private static option: OzoneAPIRequestOption = {
+		cache: true
+	}
+	/**
+	 * eventTarget to dispatch *ozone-api-request-success* and *ozone-api-request-error* events
+	 * Default value is document.
+	 * @type {EventTarget}
+	 */
+	eventTarget: EventTarget | null
+	private _resultPromise?: Promise<XMLHttpRequest>
+	private currentRequest: XMLHttpRequest | null = null
+	private resolveCurrentRequest?: { (...param: Array<any>): void }
+
+	constructor() {
+		try {
+			// should not work for nodejs
+			this.eventTarget = document
+		} catch (err) {
+			this.eventTarget = null
+		}
+	}
+
+	private _url: string = ''
+
+	get url(): string {
+		return this._url
```

Comment:
I only update this function.
the others changes are the style changes

Reply:
This is not a must-have but now that the getter only returns `_url` we could remove the `get`/`set` that adds unnecessary lines of code